src/document/DocumentManager.ts

```diff
@@ -715,14 +669,14 @@ export function select(item: SelectableItemTypes) {
 
 export async function renamePath(uuid: string, newName: string) {
   if (uiState.hasSaveLocation) {
-    const oldPath = await getTrajFilePath(uuid);
-    doc.pathlist.paths.get(uuid)?.setName(newName);
-    const newPath = await getTrajFilePath(uuid);
-    if (oldPath !== null) {
-      Promise.all([Commands.deleteFile(oldPath[0], oldPath[1])])
-        .then(() => writeTrajectory(() => newPath, uuid))
+    const traj = doc.pathlist.paths.get(uuid);
+    if (traj) {
+      await Commands.renameTraj(traj.serialize, newName)
+        .then((new_traj) => {
+          doc.pathlist.addPath(newName, true, new_traj);
```

Comment:
What does this do with the old path?

Reply:
Is just deleting the old one from the pathlist appropriate?

Reply:
Actually I don’t see the benefit of deleting and tea-adding over just changing the name. For one thing, you won’t be able to keep the same waypoint selected through the delete and readd